discord/utils.py

```diff
@@ -598,15 +602,15 @@ def _parse_ratelimit_header(request: Any, *, use_clock: bool = False) -> float:
         return float(reset_after)
 
 
-async def maybe_coroutine(f, *args, **kwargs):
+async def maybe_coroutine(f: Callable[..., Any], *args: Any, **kwargs: Any) -> Any:
```

Comment:
This might be typable? assuming a similar interface to `MaybeCoroFunc` as described before.
```suggestion
async def maybe_coroutine(f: MaybeCoroFunc[P, T], *args: P.args, **kwargs: P.kwargs) -> T:
```